### PR TITLE
refactor: all plugin use githubaction refactor

### PR DESCRIPTION
**Pull request by steinliber**
Signed-off-by: Meng JiaFeng <jiafeng.meng@merico.dev>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [ ] I have added relevant tests

## Description

- refactor all githubaction plugins, including golang, nodejs, and python
- all plugins have been tested

## Related Issues

Subtask of #850 

## New Behavior (screenshots if needed)
Current golang githubaction create method
image

### CODE REVIEW

File: internal/pkg/plugin/githubactions/golang/validate.go [hunk 1]
```diff
@@ -3,16 +3,36 @@ package golang
...
+
 // validate validates the options provided by the core.
-func validate(opts *Options) []error {
+func validateGolangStruct(opts *github.GithubActionOptions) []error {
 	retErrors := make([]error, 0)
-
 	if errs := validator.Struct(opts); len(errs) != 0 {
 		retErrors = append(retErrors, errs...)
 	}
+	if opts.Test == nil {
```
Comment (iyear) [inline internal/pkg/plugin/githubactions/golang/validate.go:33]: For structure pointers, `validate: "required"` and `ptr!=nil` are equivalent, and using the validator tag for `*Test` saves more code.

```go
type Options struct {
	Test *Test `validate:"required"`
}

type Test struct {
	Field1 string
	Field2 string
}

func main() {
	opts := Options{}
	fmt.Println(validator.New().Struct(&opts))
	// Output:
	// Key: 'Options.Test' Error:Field validation for 'Test' failed on the 'required' tag
}
```

Comment (iyear) [reply internal/pkg/plugin/githubactions/golang/validate.go:33]: All similar codes can be modified accordingly.

File: internal/pkg/plugininstaller/github/installer.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import (
+	"fmt"
+
+	"github.com/devstream-io/devstream/internal/pkg/plugininstaller"
+	"github.com/devstream-io/devstream/pkg/util/log"
+)
+
+func ProcessAction(action string) plugininstaller.BaseOperation {
+	processFunc := func(options plugininstaller.RawOptions) error {
```
Comment (iyear) [inline internal/pkg/plugininstaller/github/installer.go:11]: Why not `return func(options plugininstaller.RawOptions) error {` ?

File: internal/pkg/plugininstaller/github/option.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		Org:      opts.Org,
+		Repo:     opts.Repo,
+		NeedAuth: true,
+	}
+	return github.NewClient(ghOptions)
+}
+
+func (opts *GithubActionOptions) RenderWorkFlow(content string) (string, error) {
+	if opts.Build == nil {
+		opts.Build = &Build{false, ""}
```
Comment (iyear) [inline internal/pkg/plugininstaller/github/option.go:67]: If the default value of `Build` is zero value for each field, you can reduce this code by changing `Build *Build` to `Build Build`.

File: internal/pkg/plugininstaller/github/workflow.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package github
+
+import (
+	"github.com/devstream-io/devstream/internal/pkg/plugininstaller"
+	"github.com/devstream-io/devstream/pkg/util/github"
+)
+
+func BuildWorkFlowsWrapper(workflows []*github.Workflow) plugininstaller.MutableOperation {
+	mutFunc := func(options plugininstaller.RawOptions) (plugininstaller.RawOptions, error) {
```
Comment (iyear) [inline internal/pkg/plugininstaller/github/workflow.go:9]: Why not `return func(options plugininstaller.RawOptions) (plugininstaller.RawOptions, error) {` ?

File: pkg/util/template/render.go [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+)
+
+func Render(name, templateStr string, variable any) (string, error) {
+	t, err := template.New(name).Delims("[[", "]]").Parse(templateStr)
+	if err != nil {
+		return "", err
+	}
+
+	var buff bytes.Buffer
+	err = t.Execute(&buff, variable)
```
Comment (iyear) [inline pkg/util/template/render.go:15]: `if err=xxx;err!=nil{}` : go style

File: pkg/util/validator/validator.go [hunk 6]
```diff
@@ -29,6 +30,18 @@ func Struct(s interface{}) []error {
 	return nil
 }
 
+func StructAllError(s interface{}) error {
+	var errMsg string
+	allErrs := Struct(s)
+	if len(allErrs) > 0 {
+		for order, err := range allErrs {
+			errMsg += fmt.Sprintf("Error[%d]: %s", order, err)
```
Comment (iyear) [inline pkg/util/validator/validator.go:38]: missing `\n`

**Comment by iyear** [comment]
Very nice refactoring. Very elegant! 🚀

File: pkg/util/validator/validator.go (hunk 6 above)
Comment (iyear) [reply pkg/util/validator/validator.go:38]: This operation is not necessary. `go-playground/validator` itself returns an error, and I expanded the error to `[]error` in `Struct` func.
So this function just needs `v.Struct()`

validator error: https://github.com/go-playground/validator/blob/master/errors.go#L35-L57

example:
```go
type Options struct {
	Field1 string `validate:"required"`
	Field2 string `validate:"required"`
}

func main() {
	opts := Options{}
	fmt.Println(validator.New().Struct(&opts))
	// Output:
	// Key: 'Options.Field1' Error:Field validation for 'Field1' failed on the 'required' tag
	// Key: 'Options.Field2' Error:Field validation for 'Field2' failed on the 'required' tag
}
```

File: internal/pkg/plugin/githubactions/golang/validate.go (hunk 1 above)
Comment (iyear) [reply internal/pkg/plugin/githubactions/golang/validate.go:33]: get

**Review by iyear — APPROVED** [review]

**Review by IronCore864 — APPROVED** [review]

**Comment by IronCore864** [comment]
Great work. Merging now.

**Comment by IronCore864** [comment]
@steinliber please make sure the latest e2e-test running on the main branch after merging is successful.

**Pull request merged**